data/login/src/main/java/com/hmh/hamyeonham/login/datasource/kakao/KakaoSocialAuthDataStoreImpl.kt

```
@@ -0,0 +1,110 @@
+    override suspend fun login(): UserInfo? {
+        return try {
+            // 카카오톡이 설치되어 있으면 카카오톡으로 로그인, 아니면 계정으로 로그인
+            if (UserApiClient.instance.isKakaoTalkLoginAvailable(context)) {
+                loginWithKakaoTalk()
+            } else {
+                loginWithAccount()
+            }
+
+            // 로그인 성공 후 사용자 정보 조회
+            fetchUserInfo()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)